agent/engine/docker_task_engine.go

```diff
@@ -152,6 +166,13 @@ func (engine *DockerTaskEngine) initDockerClient() error {
 	return nil
 }
 
+func (engine *DockerTaskEngine) initS3Client() {
```

Comment:
If possible, can you move the creation of the s3 client outside the DockerTaskEngine itself and inject it?  We'd like to avoid more code that creates its own dependencies and move toward more of a dependency-injection model.